docs/api-completeness.md

```diff
@@ -0,0 +1,1217 @@
+# API Completeness
+
+Narwhals has two different level of support for libraries: "full" and "interchange".
+
+Libraries for which we have full support we intend to support the whole Narwhals API, however this is a work in progress.
+
+In the following table it is possible to check which method is implemented for which backend. 
+
+!!! info
+
+    - "pandas-like" means pandas, cuDF and Modin
+    - Polars supports all the methods (by design)
+
+| Cla | Met | pan | arr |
+| ss  | hod | das | ow  |
+|     |     | -li |     |
+|     |     | ke  |     |
+|-----|-----|-----|-----|
+| Dat | clo | :wh | :wh |
+| aFr | ne  | ite | ite |
+| ame |     | _ch | _ch |
+|     |     | eck | eck |
+|     |     | _ma | _ma |
+|     |     | rk: | rk: |
```

Comment:
this looks a bit odd here, but it renders fine:

![image](https://github.com/user-attachments/assets/ca736e7a-57c7-46aa-882a-3538325cbc7b)